### ISSUE TITLE
Problem with file upload and server rejection

### ISSUE THREAD

**Issue by jkapsouras** [paulcbetts/refit#408]
Hello,
I am using refit with xamarin and I have a problem uploading files alongside with a json with parameters in server. I made an api call like this:
`[Multipart]
[Post("/xxx/xxxx")]
Task<HttpResponseMessage> APostService([AliasAs("jsonbody")] ByteArrayPart text, [AliasAs("imagebody")] FileInfoPart filepart, [Header("Authorization")] string token);`
where the ByteArrayPart is the parameters the service posts to the server and the FileInfoPart is the image.
The above call is working on Apache Localhost but I get 403 forbidden on the nginx online server. 
The message is:
`ModSecurity: Multipart parsing error: Multipart: Invalid Content-Disposition header (-11): form-data; name=image; filename=receipt.jpg; filename*=utf-8''receipt.jpg. 
ModSecurity: Warning. Match of "eq 0" against "REQBODY_ERROR" required. [msg "COMODO WAF: The request body could not be parsed. Possibility of an impedance mismatch attack. This is not a false positive."] [data "Multipart parsing error: Multipart: Invalid Content-Disposition header (-11): form-data; name=image; filename=receipt.jpg; filename*=utf-8''receipt.jpg."]
ModSecurity: Access denied with code 403 (phase 2). Match of "eq 0" against "MULTIPART_STRICT_ERROR" required.[msg "COMODO WAF: Multipart request body failed strict validation: PE 1, BQ 1, BW 0, DB 0, DA 0, HF 0, LF 0, SM , IQ 0, IH 0, FLE 0"]`

Is it a server problem or a refit request issue? Can you help me? I have tried to send only the filearray but the problem persists.

**Comment by xljiulang** [comment]
https://github.com/xljiulang/WebApiClient/wiki/WebApiClient%E5%9F%BA%E7%A1%80
it create a standard  header like this:
name="image"

**Comment by jkapsouras** [comment]
No luck. We solved the issue by sending the image as base64 string. I still think the problem is in the headers automatically created by refit and especially here: filename=receipt.jpg; filename*=utf-8''receipt.jpg.

**Comment by onovotny** [comment]
@jkapsouras Have you tried the latest version? Mutlipart should have no issue with your example.

If you send a base64 string, then the header is correct, what you sent is a utf-8 string. (It's using StringContent internally.) If you want more control over the mime type, name parameters, etc, you can also pass in an HttpContent-derived type as well.

This looks like a server issue.

**Comment by jkapsouras** [comment]
It is most likely a server issue. It was working on localhost but not online. The online server is an nginx. I updated the package to 4.2.0 but I had some errors so I reverted back to 4.0.1. I don't remember the problem with the update. I will try it again it again when I'll have the time, the base64 suits us for now. I'll also try an ngix forum. The project is working now but I think the multipart solution is more elegant.
